torchrec/distributed/planner/planners.py

```diff
@@ -188,9 +189,7 @@ def __init__(
             )
         else:
             self._proposers = [
-                GridSearchProposer(),
-                GreedyProposer(),
-                GreedyProposer(use_depth=False),
+                DpProposer(),
```

Comment:
Let's keep default proposers the same for backward compatibility

```suggestion
                GridSearchProposer(),
                GreedyProposer(),
                GreedyProposer(use_depth=False),
```

Reply:
In all of our test scenarios, the DynamicProgrammingProposer is faster than both the GreedyProposer and GridSearchProposer. Additionally, the performance of the sharding plans generated by the DynamicProgrammingProposer is better. Which scenario might encounter issues?